rackcity/views/asset_views.py

```diff
@@ -162,8 +162,14 @@ def asset_detail(request, id):
             )
             # if id of live asset is given on a change plan, just return the corresponding related assetCP
             if assets_cp.filter(related_asset=id).exists():
+                related_asset = Asset.objects.get(id=id)
                 asset = assets_cp.get(related_asset=id)
-                serializer = RecursiveAssetCPSerializer(asset)
+                changes = get_changes_on_asset(related_asset,asset)
+                if len(changes) ==0 and not asset.is_decommissioned:
+                    asset = related_asset
+                    serializer = RecursiveAssetSerializer(asset)
+                else:
+                    serializer = RecursiveAssetCPSerializer(asset)
             elif assets_cp.filter(id=id).exists():
```

Comment:
same deal where we dont want assets with no changes to show up as on change plan in detail view